### DISCUSSION TITLE
raspotify crashs

### DISCUSSION THREAD

**Discussion by snuudz** [dtcooper/raspotify#489] [Q&A] [answered]
Hi,

I have a Raspberry Pi 3 Model B Rev 1.2 with OS Linux 10 (buster).
I put a Hifiberry Digi2 Pro on it, which works well after following this [instuction](https://www.hifiberry.com/docs/software/configuring-linux-3-18-x/).
Then, I installed **shairport-sync** by using this [simple installation instruction](https://github.com/mikebrady/shairport-sync/blob/master/INSTALL.md) and it runs perfect.
Now, I tried to install **raspotify** by using the easy way `curl -sL https://dtcooper.github.io/raspotify/install.sh | sh` 
Installation looked fine but it doesn't work.

I can select raspotify at spotify and it tells connecting. If I try to start a song, raspotify crashs.
I checked with `journalctl` and got the following:

```
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to get home directory.
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to get home directory.
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to load cookie file from cookie: No such file or directory
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to get home directory.
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to get home directory.
Jan 01 19:11:54 raspberrypi librespot[940]: Failed to load cookie file from cookie: No such file or directory
Jan 01 19:11:54 raspberrypi pulseaudio[389]: Denied access to client with invalid authentication data.
Jan 01 19:11:54 raspberrypi librespot[940]: ALSA lib pulse.c:243:(pulse_connect) PulseAudio: Unable to connect: Access denied
Jan 01 19:11:54 raspberrypi librespot[940]: [2022-01-01T18:11:54Z ERROR librespot_playback::player] Audio Sink Error Connection Refused: <AlsaSink> Device de
Jan 01 19:11:54 raspberrypi systemd[1]: raspotify.service: Main process exited, code=exited, status=1/FAILURE
Jan 01 19:11:54 raspberrypi systemd[1]: raspotify.service: Failed with result 'exit-code'.
Jan 01 19:12:04 raspberrypi systemd[1]: raspotify.service: Service RestartSec=10s expired, scheduling restart.
```

Seems something wrong with `pulseaudio`, or so.
I already searched in the net, found nothing usefull for me.

Any Idea?
Thanks a lot!

**Comment by JasonLG1979** [comment]
Simply reading the README would have answered your question.

As per the [README](https://github.com/dtcooper/raspotify#readme):

> Raspotify is a Debian package and associated repository for Raspberry Pi OS lite (Bullseye)

> Only the most current version of Raspberry Pi OS lite is supported (currently Bullseye). Raspotify may work on other OS's and/or other Raspberry Pi OS version but bugs to do with compatibility issues in unsupported systems will be closed.

> For desktop versions of Raspberry Pi OS (and/or systems with PulseAudio installed) spotifyd offers similar functionality, much better PulseAudio compatibility and is a better choice.

**Comment by snuudz** [comment]
Thanks for your hint!
I upgraded to Raspberry Pi OS lite (Bullseye), now.
I installed raspotify by using `curl -sL https://dtcooper.github.io/raspotify/install.sh | sh` without any error.
Anyway, I have the same failure at spotify.
But the error looks different with `journalctl`:
```
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to get home directory.
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to get home directory.
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to load cookie file from cookie: No such file or directory
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to get home directory.
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to get home directory.
Jan 02 21:03:10 raspberrypi librespot[1426]: Failed to load cookie file from cookie: No such file or directory
Jan 02 21:03:10 raspberrypi pulseaudio[383]: Denied access to client with invalid authentication data.
Jan 02 21:03:10 raspberrypi librespot[1426]: ALSA lib pulse.c:242:(pulse_connect) PulseAudio: Unable to connect: Access denied
Jan 02 21:03:10 raspberrypi librespot[1426]: [2022-01-02T20:03:10Z ERROR librespot_playback::player] Audio Sink Error Connection Refused: <AlsaSink> Device >
Jan 02 21:03:10 raspberrypi systemd[1]: raspotify.service: Main process exited, code=exited, status=1/FAILURE
Jan 02 21:03:10 raspberrypi systemd[1]: raspotify.service: Failed with result 'exit-code'.
Jan 02 21:03:10 raspberrypi systemd[1]: raspotify.service: Consumed 1.305s CPU time.
Jan 02 21:03:20 raspberrypi systemd[1]: raspotify.service: Scheduled restart job, restart counter is at 4.
Jan 02 21:03:20 raspberrypi systemd[1]: Stopped Raspotify (Spotify Connect Client).
Jan 02 21:03:20 raspberrypi systemd[1]: raspotify.service: Consumed 1.305s CPU time.
Jan 02 21:03:20 raspberrypi systemd[1]: Started Raspotify (Spotify Connect Client).

```

**Reply by JasonLG1979** [reply]
If only there was a [wiki](https://github.com/dtcooper/raspotify/wiki) with a [page](https://github.com/dtcooper/raspotify/wiki/Raspotify-with-PulseAudio-as-a-system-service) with a possible solution?

**Reply by JasonLG1979** [reply]
It would also be great if there were [issues in this repo that mentioned PulseAudio](https://github.com/dtcooper/raspotify/issues?q=pulseaudio)?

**Comment by snuudz** [comment, marked as answer]
It works!
Thanks a lot and sorry for bugging you.
I am a newbie to github and not familiar with it yet
